packages/block-library/src/list-item/hooks/use-indent-list-item.js

```diff
@@ -2,65 +2,41 @@
  * WordPress dependencies
  */
 import { useCallback } from '@wordpress/element';
-import { useSelect, useDispatch } from '@wordpress/data';
+import { useSelect, useDispatch, useRegistry } from '@wordpress/data';
 import { store as blockEditorStore } from '@wordpress/block-editor';
-import { createBlock, cloneBlock } from '@wordpress/blocks';
+import { createBlock } from '@wordpress/blocks';
 
 export default function useIndentListItem( clientId ) {
-	const { replaceBlocks, selectionChange, multiSelect } =
+	const registry = useRegistry();
+	const { insertBlock, moveBlocksToPosition, updateBlockListSettings } =
 		useDispatch( blockEditorStore );
 	const {
-		getBlock,
 		getPreviousBlockClientId,
-		getSelectionStart,
-		getSelectionEnd,
-		hasMultiSelection,
-		getMultiSelectedBlockClientIds,
+		getBlockRootClientId,
+		getBlockListSettings,
+		getSelectedBlockClientIds,
+		getBlockOrder,
 	} = useSelect( blockEditorStore );
 	return useCallback( () => {
-		const _hasMultiSelection = hasMultiSelection();
-		const clientIds = _hasMultiSelection
-			? getMultiSelectedBlockClientIds()
-			: [ clientId ];
-		const clonedBlocks = clientIds.map( ( _clientId ) =>
-			cloneBlock( getBlock( _clientId ) )
-		);
+		const clientIds = getSelectedBlockClientIds();
 		const previousSiblingId = getPreviousBlockClientId( clientId );
-		const newListItem = cloneBlock( getBlock( previousSiblingId ) );
-		// If the sibling has no innerBlocks, create a new `list` block.
-		if ( ! newListItem.innerBlocks?.length ) {
-			newListItem.innerBlocks = [ createBlock( 'core/list' ) ];
-		}
-		// A list item usually has one `list`, but it's possible to have
-		// more. So we need to preserve the previous `list` blocks and
-		// merge the new blocks to the last `list`.
-		newListItem.innerBlocks[
```

Comment:
This is not handled in your PR..

Trunk:

https://github.com/WordPress/gutenberg/assets/16275880/ebc6bf2a-c760-4275-8e52-e7d9ee93d8e9



This PR:

https://github.com/WordPress/gutenberg/assets/16275880/9bcaa102-ae90-4c1c-aafa-806648b2a90c

Reply:
I cannot reproduce this. My starting content:

```
<!-- wp:list -->
<ul><!-- wp:list-item -->
<li>Top<!-- wp:list -->
<ul><!-- wp:list-item -->
<li>first level</li>
<!-- /wp:list-item -->

<!-- wp:list-item -->
<li>hi<!-- wp:list -->
<ul><!-- wp:list-item -->
<li>second level</li>
<!-- /wp:list-item --></ul>
<!-- /wp:list --></li>
<!-- /wp:list-item -->

<!-- wp:list-item -->
<li><strong>Observe this item!!!</strong></li>
<!-- /wp:list-item -->

<!-- wp:list-item -->
<li>hi (2)<!-- wp:list -->
<ul><!-- wp:list-item -->
<li>second level (2)</li>
<!-- /wp:list-item --></ul>
<!-- /wp:list --></li>
<!-- /wp:list-item --></ul>
<!-- /wp:list --></li>
<!-- /wp:list-item --></ul>
<!-- /wp:list -->
```

What I see:

![list-nested-indent-outdent](https://github.com/WordPress/gutenberg/assets/4710635/2fbd3bf9-ea2c-4c50-a8e6-8cad529afeb1)